### PR TITLE
Add Terraform storage_object_lifecycle_setting sample to cgc

### PR DESCRIPTION
**Pull request by cbonnie**
If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by cbonnie** [comment]
@ScottSuarez Hey Scott, any idea why the VCR-test is failing?

**Comment by ScottSuarez** [comment]
Hey @cbonnie , are you able to access the logs? They should be available to you?

Let me know if not. If you need help parsing them let me know and we can hop on a call !

**Comment by cbonnie** [comment]
@ScottSuarez I believe every docs PR I've sent has had VCR test failures (see https://screenshot.googleplex.com/BpXrjNEW3fKvph3 as an example) and to be honest, I'm not quite sure why some of those PRs were merged but this one is considered blocking. I reviewed the logs and found an issue in line 96 here: https://screenshot.googleplex.com/A6746FeAMapBzEN, but it doesn't seem like a blocking issue.

**Comment by ScottSuarez** [comment]
> @ScottSuarez I believe every docs PR I've sent has had VCR test failures (see [screenshot.googleplex.com/BpXrjNEW3fKvph3](https://screenshot.googleplex.com/BpXrjNEW3fKvph3) as an example) and to be honest, I'm not quite sure why some of those PRs were merged but this one is considered blocking. I reviewed the logs and found an issue in line 96 here: [screenshot.googleplex.com/A6746FeAMapBzEN](https://screenshot.googleplex.com/A6746FeAMapBzEN), but it doesn't seem like a blocking issue.

This one is blocking due to the test you added failing `TestAccCGCSnippet_storageObjectLifecycleSettingExample`. I'm happy to jump on a call with you to help take a look!

**Comment by cbonnie** [comment]
> > @ScottSuarez I believe every docs PR I've sent has had VCR test failures (see [screenshot.googleplex.com/BpXrjNEW3fKvph3](https://screenshot.googleplex.com/BpXrjNEW3fKvph3) as an example) and to be honest, I'm not quite sure why some of those PRs were merged but this one is considered blocking. I reviewed the logs and found an issue in line 96 here: [screenshot.googleplex.com/A6746FeAMapBzEN](https://screenshot.googleplex.com/A6746FeAMapBzEN), but it doesn't seem like a blocking issue.
> 
> This one is blocking due to the test you added failing `TestAccCGCSnippet_storageObjectLifecycleSettingExample`. I'm happy to jump on a call with you to help take a look!

I took a look at the build log again and I saw this line:
```
    provider_test.go:309: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.
        
        (map[string]string) (len=1) {
         (string) (len=13) "force_destroy": (string) (len=5) "false"
        }
        
        
        (map[string]string) (len=1) {
         (string) (len=13) "force_destroy": (string) (len=4) "true"
        }
```

I'm not sure which part of this is causing the error. Is `force_destroy` supposed to be `true` or `false`? Is it handled differently by different tests? I think it looks like it's being ignored in some way in In https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/tests/resource_storage_bucket_test.go.erb, where I saw several instances of this:

```
				ResourceName:            "google_storage_bucket.bucket",
				ImportStateId:           fmt.Sprintf("%s/%s", getTestProjectFromEnv(), bucketName),
				ImportState:             true,
				ImportStateVerify:       true,
				ImportStateVerifyIgnore: []string{"force_destroy"}
```

Can you help me understand what's making the test fail? It'd be great if we could chat here, so I can keep record of the conversation for my own selfish learning desires, and potentially document this issue and workaround for the other collaborators on my team.

**Comment by ScottSuarez** [comment]
> > > @ScottSuarez I believe every docs PR I've sent has had VCR test failures (see [screenshot.googleplex.com/BpXrjNEW3fKvph3](https://screenshot.googleplex.com/BpXrjNEW3fKvph3) as an example) and to be honest, I'm not quite sure why some of those PRs were merged but this one is considered blocking. I reviewed the logs and found an issue in line 96 here: [screenshot.googleplex.com/A6746FeAMapBzEN](https://screenshot.googleplex.com/A6746FeAMapBzEN), but it doesn't seem like a blocking issue.
> > 
> > 
> > This one is blocking due to the test you added failing `TestAccCGCSnippet_storageObjectLifecycleSettingExample`. I'm happy to jump on a call with you to help take a look!
> 
> I took a look at the build log again and I saw this line:
> 
> ```
>     provider_test.go:309: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.
>         
>         (map[string]string) (len=1) {
>          (string) (len=13) "force_destroy": (string) (len=5) "false"
>         }
>         
>         
>         (map[string]string) (len=1) {
>          (string) (len=13) "force_destroy": (string) (len=4) "true"
>         }
> ```
> 
> I'm not sure which part of this is causing the error. Is `force_destroy` supposed to be `true` or `false`? Is it handled differently by different tests? I think it looks like it's being ignored in some way in In [https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/tests/resource_storage_bucket_test.go.erb](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/tests/resource_storage_bucket_test.go.erb?rgh-link-date=2022-08-18T15%3A35%3A32Z), where I saw several instances of this:
> 
> ```
> 				ResourceName:            "google_storage_bucket.bucket",
> 				ImportStateId:           fmt.Sprintf("%s/%s", getTestProjectFromEnv(), bucketName),
> 				ImportState:             true,
> 				ImportStateVerify:       true,
> 				ImportStateVerifyIgnore: []string{"force_destroy"}
> ```
> 
> Can you help me understand what's making the test fail? If it works with you, it'd be great if we could chat here, so I can keep record of the conversation for my own selfish learning desires, and potentially document this issue and workaround for the other collaborators on my team. :) Thanks for being patient as I learn about this code.

Ahh that makes sense! So this resource needs to be have `force_destroy` set to true in order to delete the resource that is provisioned durring the test. That's likely why the test is failing without it !

https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/storage_bucket#force_destroy

**Comment by cbonnie** [comment]
> 

I originally had `force_destroy` set to true but the test was failing, so I removed it in a previous commit, which looks like it fixed the `TestAccCGCSnippet_storageObjectLifecycleSettingExample` test that was failing before. Are you saying that I need to add the line back?

File: mmv1/products/cgc/terraform.yaml [hunk 1]
```diff
@@ -633,6 +633,13 @@ overrides: !ruby/object:Overrides::ResourceOverrides
           example_bucket_name: "example-bucket-name"
           your_topic_name: "your_topic_name"
         min_version: beta
+      - !ruby/object:Provider::Terraform::Examples
+        name: "storage_object_lifecycle_setting"
+        primary_resource_type: "google_storage_bucket"
+        primary_resource_id: "auto_expire"
```
Comment (ScottSuarez) [inline mmv1/products/cgc/terraform.yaml:639]:
```suggestion
        primary_resource_id: "auto_expire"
        ignore_read_extra:
          - "force_destroy"
```

**Review by ScottSuarez — COMMENTED** [review]
Hi bonnie,

I was in bed last night and realized what was wrong. Force destroy doesn't ned to be true as the bucket is empty! We just need to ignore read on this field for the import. Thanks so much for your patience here.

so sorry for leading you astray!

**Review by ScottSuarez — COMMENTED** [review]
Hi bonnie,

I was in bed last night and realized what was wrong. Force destroy doesn't ned to be true as the bucket is empty! We just need to ignore read on this field for the import. Thanks so much for your patience here.

so sorry for leading you astray!

**Comment by cbonnie** [comment]
Hi Scott, thanks for looking into it some more. Let's see if this works!

**Review by ScottSuarez — APPROVED** [review]
yay it passed 🎊

**Pull request merged**